disco_aws_automation/disco_deploy.py

```diff
@@ -625,9 +625,9 @@ def _generate_post_deploy_pipeline(self, pipeline_dict, old_group, ami):
         # If there is an already existing ASG, use its sizing. Otherwise, use the pipeline's sizing or a
         # reasonable default.
         if old_group:
-            desired_size = old_group.desired_capacity
-            max_size = old_group.max_size
-            min_size = old_group.min_size
+            desired_size = old_group.get('desired_capacity') or old_group['capacity']['target']
+            max_size = old_group.get('max_size') or old_group['capacity']['maximum']
```

Comment:
One option would be to create a class hierarchy for the two group type and provide an accessor method to extract the attributes. This way you could have `group.desired_capacity` which depending on the group type would execute `group.get('desired_capacity')`  for asg groups or `old_group['capacity']['target']` for spot group